src/modules/Popup/Popup.js

```diff
@@ -214,17 +213,17 @@ export default class Popup extends Component {
       return trigger
     }
 
-    const modifiers = _.merge(
-      {
-        arrow: { enabled: false },
-        flip: { enabled: !pinned },
-        // There are issues with `keepTogether` and `offset`
-        // https://github.com/FezVrasta/popper.js/issues/557
-        keepTogether: { enabled: !!offset },
-        offset: { offset },
-      },
-      popperModifiers,
-    )
+    const defaultModifiers = [
+      { name: 'arrow', enabled: false },
+      { name: 'flip', enabled: !pinned },
+      { name: 'preventOverflow', enabled: !!offset },
+      { name: 'offset', options: { offset } },
+    ]
+    const modifiers = popperModifiers
+      ? _.unionBy(popperModifiers, defaultModifiers, 'name')
```

Comment:
Merge there is not required as Popper will do it for us 😺 

https://github.com/popperjs/popper-core/blob/f1f9d1ab75b6b0e962f90a5b2a50f6cfd307d794/src/createPopper.js#L97-L99